tests/test_greedy_combination_matching.py

```diff
@@ -0,0 +1,79 @@
+from pathlib import Path
+
+import pytest
+import time
+
+from meeteval.wer.matching import greedy_combination_matching, orc_matching
+import meeteval
+from hypothesis import given, strategies as st, settings, assume
+
+example_files = (Path(__file__).parent.parent / 'example_files').absolute()
+
+# Set up hypothesis strategies
+# Limit alphabet to ensure a few correct matches. Every character represents
+# a word
+utterance = st.lists(st.integers(min_value=0, max_value=10), max_size=10)
+segments = lambda max_size: st.lists(utterance, min_size=0, max_size=max_size)
+streams = lambda max_size: st.lists(utterance, min_size=1, max_size=max_size)
+
+
+def _check_output(distance, assignment, segments, streams):
+    assert isinstance(distance, int)
+    assert distance >= 0
+    assert len(assignment) == len(segments)
+    assert all(isinstance(a, int) for a in assignment)
+    assert all(0 <= a < len(streams) for a in assignment)
+    d = orc_matching._levensthein_distance_for_assignment(segments, streams, assignment)
+    assert d == distance, (d, distance, assignment)
+
+
+@given(segments(10), streams(4))
+@settings(deadline=None)  # The tests take longer on the GitHub actions test servers
+def test_greedy_combination_matching_burn(segments, streams):
+    """Burn-test. Brute-force is exponential in the number of reference
+    utterances, so choose a small number."""
+    distance, assignment = greedy_combination_matching.greedy_combination_matching(
+        segments, streams, [0] * len(segments)
+    )
+    _check_output(distance, assignment, segments, streams)
+
+
+@given(segments(6), streams(4))
+def test_greedy_bound_by_optimal(segments, streams):
+    greedy_distance, _ = greedy_combination_matching.greedy_combination_matching(
+        segments, streams, [0] * len(segments)
+    )
+
+    optimal_distance, _ = orc_matching.orc_matching_v3(segments, streams)
+
+    assert optimal_distance <= greedy_distance
+
+
+def test_example_files_orc_equals_greedy_orc():
+    optimal_result = meeteval.wer.orcwer(example_files / 'ref.stm', example_files / 'hyp.stm')
+    greedy_result = meeteval.wer.greedy_orcwer(example_files / 'ref.stm', example_files / 'hyp.stm')
+
+    assert optimal_result == greedy_result
+
+
+@given(segments(10), streams(4), st.sampled_from(['1', '21']))
+def test_optimal_assignment_is_not_changed(segments, streams, distancetype):
+    """
+    Test that the optimal assignment is not changed.
+
+    Only works with distancetype='1'. The optimal ORC assignment is not
+    guaranteed to be optimal for other substitution costs than 1.
+    For distanctype='21' we can at least guarantee that the greedily corrected
+    assignment is not worse than the optimal assignment, but we cannot guarantee
+    equality.
```

Comment:
I just found an example where distancetype='21' changes the distance to be worse than the initialization. So, the argumentation is not correct.

Example:

```py
segments=[[2], [0, 1, 1, 1]]
streams=[[0, 0, 0, 0], [0, 1]]

# ORC assignment
assignment = [1, 0]
distance = 5

# After distancetype=2
assignment = [1, 1]
distance = 7

# After distancetype=1
assignment = [0, 1]
distance = 6  # Optimal distance of 5 is not recovered!
```

Reply:
So you plan to disable '21' for this function completely and not just partially?
I tried to guess, based on the test code and your comment.

Reply:
Hmm. Is it maybe possible to improve the greedy algorithm?
Wasn't there a paper, where `s, i, d = 4, 3, 3` was used to find an assignment?
Maybe it is worth testing that, e.g. `211 -> 433 -> 111` or `433 -> 111` instead of `211 -> 111`

Reply:
Tried that, `211 -> 433 -> 111` doesn't solve the problem for this example, `111` and `433 -> 111` solve this example correctly, but `433` does not help in all cases where `211` helps

Reply:
I'll merge this PR so that we can go forward with the DI-cpWER and greedy time-constrained variants